src/compat/should_wait_for_data_before_loaded.ts

```diff
@@ -25,7 +25,14 @@ import { isSafariMobile } from "./browser_detection";
  * @returns {Boolean}
  */
 export default function shouldWaitForDataBeforeLoaded(
-  isDirectfile : boolean
+  isDirectfile : boolean,
+  mustPlaysInline: boolean
 ) : boolean {
-  return !isDirectfile || !isSafariMobile;
+  if (isDirectfile) {
+    if (isSafariMobile && mustPlaysInline) {
+      return true;
+    }
+    return false;
+  }
+  return !(isSafariMobile && mustPlaysInline);
```

Comment:
I don't understand why the behavior changes for non-directfile contents...

Shouldn't it be more something like:
```js
return !isDirectfile || !isSafariMobile || !mustPlaysInline;
```

Reply:
Also you're definitely changing the behavior when not on Safari

Reply:
You should also probably update the tests.

Here is how it should go, if I understood correctly:
  - we're not in Directfile -> `true`
  - we're not on Safari Mobile -> `true`
  - we're on Safari Mobile without `playinline` -> `false`
  - we're on Safari Mobile with `playsinline` -> `true`

Reply:
Also, maybe you didn't understood the previous behavior because it was written in a way that is too concise.

If that's why, you can rewrite it  under a longer form, such as:

```js
if (isDirectfile && isSafariMobile) {
  return mustPlaysInline; 
}
return true;
```
Which looks more understandable.

Reply:
(Please do that if that does not look clear enough, a more readable code is a better code in that case)

Reply:
also "must plays inline" doesn't quite work as an expression.
It would be "must play inline" (without the "s")

Reply:
Thank's for the review!

Yes indeed, it was tough to wrap it in my head, was not very natural to read it like this :upside_down_face: 

So, I had difficulties to write it correctly.


> You should also probably update the tests.
> 
> Here is how it should go, if I understood correctly:
> 
> * we're not in Directfile -> `true`
> * we're not on Safari Mobile -> `true`
> * we're on Safari Mobile without `playinline` -> `false`
> * we're on Safari Mobile with `playsinline` -> `true`

Yes! you are right about the function `shouldWaitForDataBeforeLoaded`!

I will fix that :)

Reply:
I fixed the logic based on your suggestions and added/fixed tests related to the function: `shouldWaitForDataBeforeLoaded`.